Mathlib/Data/List/Infix.lean

```diff
@@ -507,4 +540,17 @@ theorem mem_of_mem_suffix (hx : a ∈ l₁) (hl : l₁ <:+ l₂) : a ∈ l₂ :=
   hl.subset hx
 #align list.mem_of_mem_suffix List.mem_of_mem_suffix
 
+theorem ne_nil_prefix {x y : List α} (hx : x ≠ []) (h : x <+: y) : y ≠ [] := by
+  rintro rfl; exact hx <| List.prefix_nil.mp h
+
+theorem get_eq_prefix {x y : List α} {n} (hn : n < x.length) (h : x <+: y) :
+    x.get ⟨n, hn⟩ = y.get ⟨n, by replace h := h.length_le; omega⟩ := by
+  obtain ⟨_, rfl⟩ := h
+  exact (List.get_append n hn).symm
+
+theorem head_eq_prefix {x y : List α} (hx : x ≠ []) (h : x <+: y) :
```

Comment:
```suggestion
theorem IsPrefix.head_eq {x y : List α} (hx : x ≠ []) (h : x <+: y) :
```

Reply:
(alternative name is `head_eq_of_isPrefix` following our naming conventions, but I think enabling the dot notation is nice.)

Reply:
If Floris agrees above, I would suggest to also swap the order of the hypotheses `hn` and `h`; putting `h` first.